lib/puppet/util/http_proxy.rb

```diff
@@ -11,6 +20,34 @@ def self.http_proxy_env
     return nil
   end
 
+  def self.no_proxy?(dest)
+    unless no_proxy_env = ENV["no_proxy"] || ENV["NO_PROXY"]
+      return false
+    end
+
+    unless dest.is_a? URI
+      begin
+        dest = URI.parse(dest)
+      rescue URI::InvalidURIError
+        return false
+      end
+    end
+
+    no_proxy_env.split(/\s*,\s*/).each do |d|
+      host, port = d.split(':')
+      host = Regexp.escape(host).gsub('\*', '.*')
+
+      regex  = port ? %r(^#{host}:#{port}$) : %r(^#{host}$)
+      string = port ? "#{dest.host}:#{dest.port}" : "#{dest.host}"
```

Comment:
Nitpick: The variable names `regex` and `string` make it hard to decipher the intent of these lines and the next.